src/git/zcl_abapgit_git_commit.clas.abap

```diff
@@ -0,0 +1,233 @@
+"! <p class="shorttext synchronized" lang="en">Git Commit</p>
+CLASS zcl_abapgit_git_commit DEFINITION
+  PUBLIC
+  CREATE PUBLIC .
+
+  PUBLIC SECTION.
+    CLASS-METHODS get_from_remote
+      IMPORTING
+        !io_repo          TYPE REF TO zcl_abapgit_repo_online
```

Comment:
Hi,
I'm trying to keep the repository concept outside of the `/git/` package, so it theoretically can be used for other things.
Suggest splitting this method into two, `get_by_branch` and `get_by_commit`, and then the URL and branch/sha1 can be passed